sdk/src/Services/DynamoDBv2/Custom/DataModel/QueryConfig.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using System.Collections.Generic;
+
+namespace Amazon.DynamoDBv2.DataModel
+{
+    /// <summary>
+    /// Input for the Query operation in the object-persistence programming model
+    /// </summary>
+#if NET8_0_OR_GREATER
+    // The DataModel namespace doesn't support trimming yet, so annotate public classes/methods as incompatible
+    [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCode(Custom.Internal.InternalConstants.RequiresUnreferencedCodeMessage)]
+#endif
+    public class QueryConfig : BaseOperationConfig
+    {
+        /// <summary>
+        /// Indicates whether a query should traverse the index backwards in descending order by range key value.
+        /// If the property is false (or not set), traversal shall be in ascending order.
+        /// </summary>
+        public bool? BackwardQuery { get; set; }
+
+        /// <summary>
+        /// Indicates the name of the index to query against.
+        /// This value is optional if the index name can be inferred from the call.
+        /// </summary>
+        public string IndexName { get; set; }
+
+        /// <summary>
+        /// The logical operator to apply to the filter conditions.
+        /// </summary>
+        /// <remarks>
+        /// <list type="bullet">
+        ///     <item>
+        ///         <term><see cref="ConditionalOperatorValues.And" /></term>
+        ///         <definition>If all of the conditions evaluate to true, then the entire filter evaluates to true.</definition>
+        ///     </item>
+        ///     <item>
+        ///         <term><see cref="ConditionalOperatorValues.Or" /></term>
+        ///         <definition>If at least one of the conditions evaluate to true, then the entire filter evaluates to true.</definition>
+        ///     </item>
+        /// </list>
+        /// The default value is <see cref="ConditionalOperatorValues.And" />.
+        /// </remarks>
+        public ConditionalOperatorValues ConditionalOperator { get; set; }
+
+        /// <summary>
+        /// Query filter for the Query operation. Evaluates the query results and returns only
+        /// the matching values. If you specify more than one condition, then by default all of the
+        /// conditions must evaluate to true. To match only some conditions, set <see cref="ConditionalOperator"/> to <see cref="ConditionalOperatorValues.Or" />.
+        /// </summary>
+        /// <remarks>
+        /// Note: Conditions must be against non-key properties.
+        /// </remarks>
+        public List<ScanCondition> QueryFilter { get; set; }
+
+        /// <summary>
+        /// Directs <see cref="DynamoDBContext" /> to ignore null values when 
+        /// converting an object to a DynamoDB item. If the property is false 
+        /// (or not set), null values will be interpreted as directives to 
+        /// delete the specific attributes on the DynamoDB item.
+        /// </summary>
+        public bool? IgnoreNullValues { get; set; }
```

Comment:
Question: Is `IgnoreNullValues` applicable to querying / scanning a DynamoDB table? The existing documentation makes me believe I'd only have to use this property when saving items: https://github.com/aws/aws-sdk-net/blob/e689b8f37366972e04ef2c51892f277ff438e1c9/sdk/src/Services/DynamoDBv2/Custom/DataModel/Configs.cs#L104-L110

Reply:
When I looked at this closer, there's a potential call path that still looks at `IgnoreNullValues` through either `ContextInternal.ComposeQueryFilter` (applicable to Query/Scan) and `ContextInternal.MakeKey` (Delete/Load)`

Both of those methods need to convert the user-provided hash or range key from its .NET type our `DynamoDBEntry` that we'll use to build the low-level requests.

This calls into this stack:

```
ContextInternal.ValueToDynamoDBEntry
ContextInternal.ToDynamoDBEntry
ContextInternal.SerializeToDocument
ContextInternal.ObjectToItemStorageHelper
ContextInternal.PopulateItemStorage
ContextInternal.ShouldSave
```

Then here we're looking at `IgnoreNullValues`: https://github.com/aws/aws-sdk-net/blob/fcdebdc0e8859bc8d0317e0130da01c1ae3f0b67/sdk/src/Services/DynamoDBv2/Custom/DataModel/ContextInternal.cs#L299-L305

This is only relevant when:
1. You're using a key in Load/Delete, or as a query or scan filter that is `null`.
2. Its type doesn't have a custom conversion applied
3. Its type doesn't have a built-in converter applied via either V1 or V2.

Reply:
After playing with this for a few more hours, I'm not finding a case where `IgnoreNullValues` is influencing the scan or query filter. Tried null values types, null collections, collections with null elements.

Reply:
Removed in https://github.com/aws/aws-sdk-net/pull/3383/commits/987fe702d9428657cff4228bf91457a1ae9dcd74